cmd/relayproxy/service/gofeatureflag.go

```diff
@@ -159,92 +162,118 @@ func initRetriever(c *config.RetrieverConf) (retriever.Retriever, error) {
-		FlushInterval: func() time.Duration {
-			if c.FlushInterval != 0 {
-				return time.Duration(c.FlushInterval) * time.Millisecond
-			}
-			return config.DefaultExporter.FlushInterval
-		}(),
-		MaxEventInMemory: func() int64 {
-			if c.MaxEventInMemory != 0 {
-				return c.MaxEventInMemory
-			}
-			return config.DefaultExporter.MaxEventInMemory
-		}(),
-	}
 
 	switch c.Kind {
 	case config.WebhookExporter:
-		dataExp.Exporter = &webhookexporter.Exporter{
-			EndpointURL: c.EndpointURL, Secret: c.Secret, Meta: c.Meta, Headers: c.Headers}
-		return dataExp, nil
+		return &webhookexporter.Exporter{
+			EndpointURL: c.EndpointURL,
+			Secret:      c.Secret,
+			Meta:        c.Meta,
+			Headers:     c.Headers,
+		}, nil
 	case config.FileExporter:
-		dataExp.Exporter = &fileexporter.Exporter{Format: format, OutputDir: c.OutputDir, Filename: filename,
-			CsvTemplate: csvTemplate, ParquetCompressionCodec: parquetCompressionCodec}
-		return dataExp, nil
+		return &fileexporter.Exporter{
+			Format:                  format,
+			OutputDir:               c.OutputDir,
+			Filename:                filename,
+			CsvTemplate:             csvTemplate,
+			ParquetCompressionCodec: parquetCompressionCodec,
+		}, nil
 	case config.LogExporter:
-		dataExp.Exporter = &logsexporter.Exporter{
+		return &logsexporter.Exporter{
 			LogFormat: func() string {
 				if c.LogFormat != "" {
 					return c.LogFormat
 				}
 				return config.DefaultExporter.LogFormat
 			}(),
-		}
-		return dataExp, nil
+		}, nil
 	case config.S3Exporter:
 		awsConfig, err := awsConf.LoadDefaultConfig(context.Background())
 		if err != nil {
-			return dataExp, err
+			return nil, err
 		}
-		dataExp.Exporter = &s3exporterv2.Exporter{
+
+		return &s3exporterv2.Exporter{
 			Bucket:                  c.Bucket,
 			Format:                  format,
 			S3Path:                  c.Path,
 			Filename:                filename,
 			CsvTemplate:             csvTemplate,
 			ParquetCompressionCodec: parquetCompressionCodec,
 			AwsConfig:               &awsConfig,
-		}
-		return dataExp, nil
+		}, nil
 	case config.GoogleStorageExporter:
-		dataExp.Exporter = &gcstorageexporter.Exporter{
+		return &gcstorageexporter.Exporter{
 			Bucket:                  c.Bucket,
 			Format:                  format,
 			Path:                    c.Path,
 			Filename:                filename,
 			CsvTemplate:             csvTemplate,
 			ParquetCompressionCodec: parquetCompressionCodec,
-		}
-		return dataExp, nil
+		}, nil
 	case config.SQSExporter:
 		awsConfig, err := awsConf.LoadDefaultConfig(context.Background())
 		if err != nil {
-			return dataExp, err
+			return nil, err
 		}
-		dataExp.Exporter = &sqsexporter.Exporter{QueueURL: c.QueueURL, AwsConfig: &awsConfig}
-		return dataExp, nil
 
+		return &sqsexporter.Exporter{
+			QueueURL:  c.QueueURL,
+			AwsConfig: &awsConfig,
+		}, nil
+	case config.KafkaExporter:
+		return &kafkaexporter.Exporter{
+			Format:   format,
+			Settings: c.Kafka,
+		}, nil
```

Comment:
During my tests I've seen that the `Settiings` key has always empty value for the fields.
It may be because there are no annotations for the fields in the retriever? 

This is the configuration file I use for the relayproxy:
```yaml
listen: 1031
pollingInterval: 1000
retriever:
  kind: file
  path: ./openfeature/provider_tests/flags.yaml
exporter:
  kind: kafka
  format: json
  settings:
    topic: go-feature-flag-events
    addresses:
      - localhost:29092
enableSwagger: true
```

Reply:
The configuration struct expects the `kafka` key for the settings. I'm not 100% sure about the name though, since it might be a bit repeitive. Using `settings` is too generic too. Maybe `kafka_settings`?